### PR TITLE
fix compilation

### PR DESCRIPTION
**Pull request by mtrive**
### Description
Fix backwards compatibility with old versions of Unity.

### CODE REVIEW

File: Editor/ProjectAuditorSettingsProvider.cs [hunk 1]
```diff
@@ -106,9 +106,10 @@ public void SelectCurrentSettings(ProjectAuditorSettings settings)
             m_CurrentSettings = settings;
 
             var path = AssetDatabase.GetAssetPath(settings);
-            var guid = AssetDatabase.GUIDFromAssetPath(path);
+            var guidAsString = AssetDatabase.AssetPathToGUID(path);
 
-            var guidAsString = path == "" || guid.Empty() ? "" : guid.ToString();
+            if (!string.IsNullOrEmpty(guidAsString))
+                UserPreferences.settingsAsset = guidAsString;
             UserPreferences.settingsAsset = guidAsString;
```
Comment (borisbauer-unity) [inline Editor/ProjectAuditorSettingsProvider.cs:113]: I see that this line matches the conditional line just above.

So the intention here was:  If the GUID is not valid (the else-case basically) set the preferences to empty string "" to signal the fallback to default (in-memory) settings.

**Comment by borisbauer-unity** [comment]
Seems to work well now. The change is more a clean-up:  just removed one conditional line that wrote into UserPreferences.

What I expect as a user and tested:

- I select settings and they stay a persistent selection (Default or a settings asset)
- if I delete a selected settings asset while PA is closed or running it falls back onto the in-memory default settings

**Comment by mtrive** [comment]
> Seems to work well now. The change is more a clean-up: just removed one conditional line that wrote into UserPreferences.
> 
> What I expect as a user and tested:
> 
> * I select settings and they stay a persistent selection (Default or a settings asset)
> * if I delete a selected settings asset while PA is closed or running it falls back onto the in-memory default settings

Great, thank you!

File: Editor/ProjectAuditorSettingsProvider.cs (hunk 1 above)
Comment (mtrive) [reply Editor/ProjectAuditorSettingsProvider.cs:111]: You ended up making the same change I had locally :-)
Feel free to approve and merge.

**Pull request merged**